packages/dynamic_layouts/lib/src/wrap_layout.dart

```diff
@@ -0,0 +1,270 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:dynamic_layouts/dynamic_layouts.dart';
+import 'package:flutter/rendering.dart';
+
+// The model that tracks the current max size of the Sliver in the mainAxis and
+// tracks if there is still space on the crossAxis.
+class _RunMetrics {
+  _RunMetrics({
+    required this.maxSliver,
+    required this.currentSizeUsed,
+    required this.scrollOffset,
+  });
+
+  double maxSliver;
+  double currentSizeUsed;
+  double scrollOffset;
+}
+
+/// Controls the layout of tiles in a grid.
+/// A [SliverGridLayout] that uses dynamically sized tiles.
```

Comment:
https://dart.dev/guides/language/effective-dart/documentation#do-start-doc-comments-with-a-single-sentence-summary

The first paragraph should be a single sentence summary.